utils/interfaces/_library/appsec_iast.py

```diff
@@ -28,32 +26,22 @@ def check(self, data):
 
             for i, span in enumerate(get_spans_related_to_rid(content, self.rid)):
                 self.log_debug(f'Found span with rid={self.rid}: span_id={span["span_id"]}')
-                self.spans.append(f'{span["trace_id"]}#{span["span_id"]}')
 
-                # I would like to make validations even if there aren't iast events
-                # if "_dd.iast.json" in span.get("meta", {}):
-                self.appsec_iast_events.append({"span": span, "i": i, "log_filename": data["log_filename"]})
-
-    def _get_related_spans(self):
-        return [event for event in self.appsec_iast_events if "span" in event]
+                if "_dd.iast.json" in span.get("meta", {}):
+                    self.appsec_iast_events.append({"span": span, "i": i, "log_filename": data["log_filename"]})
 
     def final_check(self):
-        spans = self._get_related_spans()
-
-        def vulnerabilityDecoder(vulDict):
+        def vulnerability_decoder(vulDict):
             return namedtuple("X", vulDict.keys())(*vulDict.values())
 
-        for span in spans:
-            if not self.closed:
-                span_data = span["span"]
-                vulnerabilities = []
-                if "_dd.iast.json" in span_data["meta"].keys():
-                    appsec_iast_data = json.loads(span_data["meta"]["_dd.iast.json"], object_hook=vulnerabilityDecoder)
-                    vulnerabilities = appsec_iast_data.vulnerabilities
+        for span in self.appsec_iast_events:
+            if "span" in span:
```

Comment:
useless, you control this object, and `span` is always in it. Or am I missing something ?